TombEngine/Scripting/Internal/TEN/Collision/Probe.cpp

```diff
@@ -0,0 +1,329 @@
+#include "framework.h"
+#include "Scripting/Internal/TEN/Collision/Probe.h"
+#include "Scripting/Internal/TEN/Collision/MaterialTypes.h"
+
+#include "Game/collision/Point.h"
+#include "Game/Lara/lara_climb.h"
+#include "Scripting/Internal/LuaHandler.h"
+#include <Scripting/Internal/ReservedScriptNames.h>
+#include "Scripting/Internal/TEN/Objects/Moveable/MoveableObject.h"
+#include "Scripting/Internal/TEN/Objects/Room/RoomObject.h"
+#include "Scripting/Internal/TEN/Types/Vec3/Vec3.h"
+#include "Scripting/Internal/TEN/Types/Rotation/Rotation.h"
+#include "Specific/level.h"
+
+using namespace TEN::Collision::Point;
+
+namespace TEN::Scripting::Collision
+{
+	/// Represents a collision probe in the game world.
+	// Provides collision information from a reference world position.
+	//
+	// @tenclass Collision.Probe
+	// pragma nostrip
+
+	void Probe::Register(sol::table& parent)
+	{
+		using ctors = sol::constructors<
+			Probe(const Vec3&, int),
+			Probe(const Vec3&, int, const Vec3&, float),
+			Probe(const Vec3&, int, const Rotation&, float),
+			Probe(const Vec3&, int, const Rotation&, const Vec3&)>;
+
+		// Register type.
+		parent.new_usertype<Probe>(
+			ScriptReserved_Probe,
+			ctors(), sol::call_constructor, ctors(),
+
+			// Getters
+			ScriptReserved_GetColPosition, &Probe::GetPosition,
+			ScriptReserved_GetColRoomNumber, &Probe::GetRoomNumber,
+			ScriptReserved_GetColRoomName, &Probe::GetRoomName,
+			ScriptReserved_GetColRoom, & Probe::GetRoom,
+			ScriptReserved_GetFloorHeight, &Probe::GetFloorHeight,
+			ScriptReserved_GetCeilingHeight, &Probe::GetCeilingHeight,
+			ScriptReserved_GetWaterSurfaceHeight, &Probe::GetWaterSurfaceHeight,
+			ScriptReserved_GetFloorNormal, &Probe::GetFloorNormal,
+			ScriptReserved_GetCeilingNormal, &Probe::GetCeilingNormal,
+			ScriptReserved_GetFloorMaterialType, &Probe::GetFloorMaterialType,
+			ScriptReserved_GetCeilingMaterialType, &Probe::GetCeilingMaterialType,
+
+			// Inquirers
+			ScriptReserved_IsSteepFloor, &Probe::IsSteepFloor,
+			ScriptReserved_IsSteepCeiling, &Probe::IsSteepCeiling,
+			ScriptReserved_IsWall, &Probe::IsWall,
+			ScriptReserved_IsInsideSolidGeometry, &Probe::IsInsideSolidGeometry,
+			ScriptReserved_IsClimbableWall, &Probe::IsClimbableWall,
+			ScriptReserved_IsMonkeySwing, &Probe::IsMonkeySwing,
+			ScriptReserved_IsDeathTile, &Probe::IsDeath);
+	}
+
+	/// Create a Probe at a specified world position in a room.
+	// @function Probe
+	// @tparam Vec3 pos World position.
+	// @tparam int roomNumber Room number.
+	// @treturn Probe A new Probe.
+	Probe::Probe(const Vec3& pos, int roomNumber)
```

Comment:
I suggest to make `roomNumber` optional here, and in case it is missing, fetch it using `FindRoomNumber`.